packages/kit/src/core/postbuild/prerender.js

```diff
@@ -459,19 +460,20 @@ async function prerender({ out, manifest_path, metadata, verbose, env }) {
 
 					if (processed_id.includes('[')) continue;
 					const path = `/${get_route_segments(processed_id).join('/')}`;
-					enqueue(null, config.paths.base + path);
+					entries_to_enqueue.push({ path: config.paths.base + path });
 				}
 			}
 		} else {
-			enqueue(null, config.paths.base + entry);
+			entries_to_enqueue.push({ path: config.paths.base + entry });
 		}
 	}
 
 	for (const { id, entries } of route_level_entries) {
 		for (const entry of entries) {
-			enqueue(null, config.paths.base + entry, undefined, id);
+			entries_to_enqueue.push({ path: config.paths.base + entry, id });
 		}
 	}
+	await Promise.all(entries_to_enqueue.map(({ path, id }) => enqueue(null, path, undefined, id)));
```

Comment:
why is this better than what existed before this PR?

Reply:
it's clearer to me as it's explicit that we intend to enqueue them in parallel. it also handles rejected promise errors which would not be handled before. while it's a pretty simple method that's unlikely to fail, but everytime I come across something like that in a code review I have to review it and assess the risk that it might fail and it's easier for me to not have to do that

Reply:
Rejected promise errors _are_ handled already. That's what the `await q.done()` two lines later is for. This is duplicative and less efficient, it adds nothing!